lib/srv/db/auth/auth.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/gravitational/teleport/lib/srv/db/session"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/teleport/lib/utils"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/service/rds/rdsutils"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+)
+
+// AuthenticatorConfig is the database access authenticator configuration.
+type AuthenticatorConfig struct {
+	// AuthClient is the cluster auth client.
+	AuthClient *auth.Client
+	// Credentials are the AWS credentials used to generate RDS auth tokens.
+	Credentials *credentials.Credentials
+	// RDSCACerts contains AWS RDS root certificates.
+	RDSCACerts map[string][]byte
+	// Clock is the clock implementation.
+	Clock clockwork.Clock
+}
+
+// CheckAndSetDefaults validates the config and sets defaults.
+func (c *AuthenticatorConfig) CheckAndSetDefaults() error {
+	if c.AuthClient == nil {
+		return trace.BadParameter("missing AuthClient")
+	}
+	if c.Credentials == nil {
+		return trace.BadParameter("missing Credentials")
+	}
+	if c.Clock == nil {
+		c.Clock = clockwork.NewRealClock()
+	}
+	return nil
+}
+
+// Authenticator provides utilities for authenticating and authorizing access
```

Comment:
nit: so this is no longer an authenticator. How about access validator or checker?